Dependencies/UrlLib/Source/Unix/UrlRequest.cpp

```diff
@@ -37,22 +37,21 @@ namespace UrlLib
 
         arcana::task<void, std::exception_ptr> SendAsync()
         {
-            return arcana::make_task(arcana::threadpool_scheduler, m_cancellationSource, [this]()
+            m_taskCompletionSource = {};
```

Comment:
If multiple requests are active at the same time, does it work?

Reply:
I've tested a loop with 10 simultaneous HTTPS requests and it worked fine.

Reply:
I think what we need to do is associate some information with the CURL handle. This includes the cancellation source and the task completion source. Can be a separate issue though.